.github/workflows/testing.yml

```diff
@@ -12,9 +12,25 @@ jobs:
       - run: echo "The job was automatically triggered by a ${{ github.event_name }} event."
       - run: echo "The name of your branch is ${{ github.ref }} and your repository is ${{ github.repository }}."
       - run: echo "The ${{ github.repository }} repository has been cloned to the runner."
-      - run: pip install .
-      - run: pip install pytest
-      - run: pytest -v puma/tests/
+      - run: pip install -e . # install in editable mode for coverage to work
+      - run: pip install flake8 pytest pytest-cov
+      - run: pytest -v --junitxml=pytest.xml --cov-report=term-missing:skip-covered --cov=puma puma/tests/ | tee pytest-coverage.txt
+      - name: Pytest coverage comment
+        uses: MishaKav/pytest-coverage-comment@main
+        with:
+          pytest-coverage-path: ./pytest-coverage.txt
+          junitxml-path: ./pytest.xml
+      # For commits on the main branch, check if the
+      - name: Update README with Coverage Html
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: |
+          sed -i '/<!-- Pytest Coverage Comment:Begin -->/,/<!-- Pytest Coverage Comment:End -->/c\<!-- Pytest Coverage Comment:Begin -->\n\${{ steps.coverageComment.outputs.coverageHtml }}\n${{ steps.coverageComment.outputs.summaryReport }}\n<!-- Pytest Coverage Comment:End -->' ./README.md
+      - name: Commit & Push changes to README
+        if: ${{ github.ref == 'refs/heads/main' }}
+        uses: actions-js/push@master
+        with:
+          message: Update coverage on Readme
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
this adds the coverage badge to the README for commits on main (I used the instructions from [here](https://github.com/marketplace/actions/pytest-coverage-comment).

But I think this creates quite a messy README file?

What do you think @manuguth ?

Reply:
hm ok, that's not very practical to push on the main branch - which we actually disabled 

there is no generic url we can access the coverage as it is possible on gitlab?

Reply:
not that I'm aware of. 

Let's skip the README part for now and take care of that in another PR?